src/parser/Parser.ts

```diff
@@ -832,45 +836,49 @@ export class Parser {
                     );
                 }
 
-                let blockEnd = peek();
+                let blockEnd = maybeThenBranch.closingToken;
                 if (blockEnd.kind === Lexeme.EndIf) {
                     endIfToken = blockEnd;
                 }
 
-                thenBranch = maybeThenBranch;
+                thenBranch = maybeThenBranch.body;
 
                 // attempt to read a bunch of "else if" clauses
-                while (check(Lexeme.ElseIf)) {
-                    elseIfTokens.push(advance());
+                while (blockEnd.kind === Lexeme.ElseIf) {
+                    elseIfTokens.push(blockEnd);
                     let elseIfCondition = expression();
                     if (checkThen()) {
                         // `then` is optional after `else if ...condition...`, so only advance to the next token if `then` is present
                         advance();
                     }
 
-                    let elseIfThen = block(Lexeme.EndIf, Lexeme.Else, Lexeme.ElseIf);
-                    if (!elseIfThen) {
+                    let maybeElseIfThen = block(Lexeme.EndIf, Lexeme.Else, Lexeme.ElseIf);
+                    if (!maybeElseIfThen) {
                         throw addError(
                             peek(),
                             "Expected 'end if', 'else if', or 'else' to terminate 'then' block"
                         );
                     }
 
-                    let blockEnd = previous();
+                    blockEnd = maybeElseIfThen.closingToken;
                     if (blockEnd.kind === Lexeme.EndIf) {
                         endIfToken = blockEnd;
                     }
 
                     elseIfBranches.push({
                         type: "ElseIf",
                         condition: elseIfCondition,
-                        thenBranch: elseIfThen,
+                        thenBranch: maybeElseIfThen.body,
                     });
                 }
 
-                if (match(Lexeme.Else)) {
-                    elseBranch = block(Lexeme.EndIf);
-                    endIfToken = advance();
+                if (blockEnd.kind === Lexeme.Else) {
+                    let maybeElseBranch = block(Lexeme.EndIf);
+                    if (!maybeElseBranch) {
+                        throw addError(peek(), "Expected 'end if' to terminate 'else' block");
+                    }
+                    elseBranch = maybeElseBranch.body;
+                    endIfToken = maybeElseBranch.closingToken;
```

Comment:
Ooh nice does this mean the `endIfToken` is populated now? I ran into this when I was writing the coverage tool

Reply:
It does!  That's where a few of the snapshot updates came into play 😃